packages/@expo/cli/src/verifyNativeAssets/verifyNativeAssets.ts

```diff
@@ -0,0 +1,172 @@
+  if (buildManifestPaths.length === 0) {
+    throw new CommandError(`No app.manifest found in build path`);
+  }
+  const buildManifestPath = buildManifestPaths[0];
+  debug(`Build manifest found at ${buildManifestPath}`);
+  const buildManifestString = fs.readFileSync(buildManifestPaths[0], { encoding: 'utf-8' });
+  const buildManifest: BuildManifest = JSON.parse(buildManifestString);
+  return buildManifest;
+}
+
+/**
+ * Extracts the set of asset hashes from a build manifest.
+ *
+ * @param buildManifest The build manifest
+ * @returns The set of asset hashes contained in the build manifest
+ */
+export function getBuildManifestHashSet(buildManifest: BuildManifest) {
+  return new Set((buildManifest.assets ?? []).map((asset) => asset.packagerHash));
+}
+
+/**
+ * Reads and extracts the asset dump for an exported bundle.
+ *
+ * @param exportPath The path to the exported bundle containing an asset dump.
+ * @returns The asset dump as an object.
+ */
+export function getFullAssetDump(exportPath: string) {
+  const assetMapPath = path.resolve(exportPath, 'assetmap.json');
+  if (!fs.existsSync(assetMapPath)) {
+    throw new CommandError(
+      `The export bundle chosen does not contain assetmap.json. Please generate the bundle with "npx expo export --dump-assetmap"`
+    );
+  }
+  const assetMapString = fs.readFileSync(assetMapPath, { encoding: 'utf-8' });
+  const assetMap: FullAssetDump = new Map(Object.entries(JSON.parse(assetMapString)));
+  return assetMap;
+}
+
+/**
+ * Extracts the set of asset hashes from an asset dump.
+ *
+ * @param assetDump
+ * @returns The set of asset hashes in the asset dump
+ */
+export function getFullAssetDumpHashSet(assetDump: FullAssetDump) {
+  const assetSet = new Set<string>();
+  assetDump.forEach((_asset, hash) => {
+    assetSet.add(hash);
+  });
+  return assetSet;
+}
+
+/**
+ * Reads and extracts the metadata from an exported bundle.
+ *
+ * @param exportPath Path to the exported bundle.
+ * @returns The metadata of the bundle.
+ */
+export function getExportedMetadata(exportPath: string) {
+  const metadataPath = path.resolve(exportPath, 'metadata.json');
+  if (!fs.existsSync(metadataPath)) {
+    throw new CommandError(
+      `The export bundle chosen does not contain metadata.json. Please generate the bundle with "npx expo export --dump-assetmap"`
+    );
+  }
+  const metadataString = fs.readFileSync(metadataPath, { encoding: 'utf-8' });
+  const metadata: ExportedMetadata = JSON.parse(metadataString);
+  return metadata;
+}
+
+/**
+ * Extracts the set of asset hashes from an exported bundle's metadata for a given platform.
+ *
+ * @param metadata The metadata from the exported bundle
+ * @param platform Either 'android' or 'ios'
+ * @returns the set of asset hashes
+ */
+export function getExportedMetadataHashSet(metadata: ExportedMetadata, platform: Platform) {
+  const fileMetadata =
+    platform === 'android' ? metadata.fileMetadata.android : metadata.fileMetadata.ios;
```

Comment:
Same here

Reply:
Since platform is now required, it is probably safe to leave this in